### ISSUE TITLE
Renaming / Naming bookmarks

### ISSUE THREAD

**Issue by selim** [quran/quran-ios#541]
Please add a rename button beside the delete button in bookmarks. When user click the rename button and enter the name, show that name at beside (left/right) of current bookmark label. 

It's will be great benefit for readers following multiple (completion) positions for reciting Quran same time.

**Comment by mohamede1945** [comment]
Salam alikom,

Ok, just to understand the use case. Someone would like to complete the Quran multiple times simultaneously. So, they need a way to track

**Comment by mohamede1945** [comment]
Salam alikom,

Just to ensure I've understood the scenario correctly: A user aims to complete multiple readings of the Quran concurrently, and hence, requires a method to track their progress in each reading. One way to do that is to use bookmarks. So, your suggestion is to attach labels to these bookmarks, allowing users to easily identify which Quran reading each bookmark corresponds to, correct?

If that's correct, maybe we need to brainstorm multiple ideas.

**Comment by selim** [comment]
Alaikum salam,

Yes correct...  
One user aims to complete concurrently with  multiple readings same time,

Use cases:
1. User can be contribute reading groups and complete assigments, a sample image Juz sharing group in turkiye:
<img width="489" alt="image" src="https://github.com/quran/quran-ios/assets/241385/4afaf138-32f9-4e88-a7d0-eaa05dd36965">
2. Same time user aims to complete multiple readings concurrently for different purposes,
3. Each completion must be follow independently, may be scheduled.

an idea ->
- users can add names to bookmarks, 
- While reading, when user click the bookmark button,  app show the bookmark list, and user can select to update related bookmark.

**Comment by mohamede1945** [comment]
I believe it's a perfectly valid use case. However, I'm still uncertain if utilizing named bookmarks is the optimal solution. I've been considering the development of a feature called 'Completions'. This would allow users to initiate and track multiple Quran reading completions. Each completion could be optionally named, and the system would monitor their progress. Additionally, it would provide users with extra details about their reading journey, such as the amount remaining to read, which specific pages they haven't tackled yet, when they started each completion, as well as their average time spent on each page, among other insights.

In terms of future enhancements to this feature, we could incorporate Quran daily verses (Daily Werd) into the completions, with daily notifications.

What are your thoughts on this?

**Comment by selim** [comment]
Perfecto; Completion's feature which you mentioned is what we need. But please, we do not want to wait for 1 year. All users will be happy if released asap.

**Comment by mohamede1945** [comment]
I'll do my best. May Allah help me finish it soon.

**Comment by selim** [comment]
I saw many issues would be solved with "Completion Feature". May be we can review old requests and merge with "Completion Feature" and develop just one sprint.

**Comment by selim** [comment]
Any iterations for "Completion Feature" ?

**Comment by mohamede1945** [comment]
No, I'm currently traveling, will be back mid October. Plus, I have other in-progress features that I'll be focusing on when I return. This feature is open for contribution, so feel free to jump in! :)

**Comment by AhmedG** [comment]
I think this would work well with a scheduling system (reminder to read).

**Comment by selim** [comment]
@mohamede1945 any schedule for that?

**Comment by mohamede1945** [comment]
Salam alikom, we are releasing the quran.com sync feature next month inshaa'Allah. There is a feature called "Reading Bookmarks" that I think is very similar to what you need. Could you try `make run-example-sync` locally and see if this satisfies your requirement. It's basically a bookmark that's movable - it has a single location and user can only move it from one location to another.

**Comment by selim** [comment]
Wa alaikum assalam

Thanks! I'll test it locally. Once the quran.com sync feature is ready, I'd also be happy to contribute to the project inshaa'Allah.

**Comment by mohamede1945** [comment]
I think it's ready for initial testing, there are a few P2 fixes I'm working on right now. But you can test it locally without any sync impl for now.

Also if you would like to contribute, just let me know what you are interested in working on please.

Jazak Allah khyrn

**Comment by selim** [comment]
I'm testing now,, is collections limited with 3 bookmarks? 

<img width="417" height="315" alt="Image" src="https://github.com/user-attachments/assets/84424792-de41-4b35-a73c-c9112be7a1ea" />

Great job on the Bookmarks feature—it's very useful! It looks like Collections is still under development, as the new collection I created but bookmarks couldn't added to Collection. Also, should collections be selected from Collections menu (main menu) rather than directly from the Bookmark menu while reading?
